mathesar/utils/datafiles.py

```diff
@@ -114,13 +114,18 @@ def create_datafile(data):
         )
     else:
         max_level = data.get('max_level', 0)
+        sheet_name = data.get('sheet_name', '1')
+        # Pandas indexes sheets from 0 and not 1.
+        if sheet_name.isdigit():
+            sheet_name = str(int(sheet_name) - 1)
         datafile = DataFile(
             file=raw_file,
             base_name=base_name,
             type=type,
             created_from=created_from,
             header=header,
-            max_level=max_level
+            max_level=max_level,
```

Comment:
Argh, this is a parameter for JSON parsing, right? I do hope we get to fix all of this logic being lumped together during your project @IamEzio .

Reply:
Yes @dmos62, this is one of our postponed task: #3030. We'll create different models for different types of datafile each extending the current DataFile class with only the essential attributes.